test/integration/collection-management/collection.test.ts

```diff
@@ -422,6 +429,41 @@ describe('Collection', function () {
     });
   });
 
+  describe('#createIndex', () => {
```

Comment:
Moved this out of unit tests so it could be configured properly / not default to OpQuery incorrectly. We can also make `mockServer` default to a specific `maxWireVersion >= 7` to stop this error, but I figured since it's only 2 tests for now, its not worth the effort.